src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -398,6 +413,14 @@ export default class QueryStringInputUI extends Component<Props, State> {
       this.setState({ isSuggestionsVisible: false, index: null });
       this.onSubmit({ query: newQueryString, language: this.props.query.language });
     }
+
+    if (
+      this.props.autoSubmit &&
```

Comment:
Could this block be merged with the one above it? The content of the `if` blocks is the same.

```
if (
 type === QuerySuggestionTypes.RecentSearch ||
 this.props.autoSubmit &&
      (type === QuerySuggestionTypes.Value || [':*', ': *'].includes(value.trim())
  ) {
```

Also, not sure what this condition means `[':*', ': *'].includes(value.trim())` ?

Reply:
i merged both blocks.

So that condition make sures if user selects all operator :*, in that case it should also submit the request.